scripts/changed-files

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e
+[ $# -ne 1 ] && echo "Usage: $0 DIR" >&2 && exit 1
+cd "$1"
+STAGED_UNSTAGED=$(git diff --name-only HEAD)
+[ -n "$STAGED_UNSTAGED" ] && echo "$STAGED_UNSTAGED" && exit
```

Comment:
Why this part? Doesn't `git ls-files` handle this case too? In fact, if this isn't needed, the whole script can be replaced with `git -C ... ls-files`,

Reply:
Premature optimization…